File: baekjoon/bj10773.cpp
/*
문제:백준10773번
출처:https://www.acmicpc.net/problem/10773
*/
#include<iostream>
#include<stack>
#include<string>
using namespace std;

int main() {
	stack<int>s;
	int N, a;
	int sum = 0;
	cin >> N;
	while (N--) {
		cin >> a;
		if (a == 0) {
			sum -= s.top();
			s.pop();
		}
		else {
			s.push(a);
			sum += a;
		}
	}
	cout << sum << '\n';
}

File: baekjoon/bj10773.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Iterator;
import java.util.Stack;
import java.util.StringTokenizer;

public class bj10773 {
    static Stack<Integer> st = new Stack<Integer>();
    static int K;

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        K = Integer.parseInt(br.readLine());

        for(int i=0;i<K;i++){
            int num = Integer.parseInt(br.readLine());
            if(num!=0) st.push(num);
            else{
                if(!st.isEmpty()) st.pop();
            }
        }

        Iterator<Integer> it = st.iterator();
        int sum = 0;
        while(it.hasNext()){
            sum += it.next();
        }
        System.out.println(sum);
        br.close();
    }
}
